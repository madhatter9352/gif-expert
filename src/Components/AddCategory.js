import React, {useState} from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCateg}) => {
    const [inputValue, setInputValue] = useState("")

    const handleInputValue = (e) => {
        setInputValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCateg(cate => [inputValue, ...cate]);
            setInputValue("");
        }

    }

    return (
        <form onSubmit = { handleSubmit }>
            <div>
                <input 
                    type = "text"
                    value = { inputValue }
                    onChange = { handleInputValue }
                />
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCateg: PropTypes.func.isRequired 
}

export default AddCategory
