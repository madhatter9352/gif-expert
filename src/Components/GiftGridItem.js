import React from 'react'
import PropTypes from 'prop-types';

export const GiftGridItem = ( {id, title, url} ) => {
    console.log(id, title, url);
    return (
        <div className="card animate__animated animate__bounce">
            <img src = {url} alt={title} width={250} style={{padding: 10}}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
