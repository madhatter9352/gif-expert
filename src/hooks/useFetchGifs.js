import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGift';

export const useFectchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category)
            .then(img => {
                setTimeout(() => {
                    setState({
                        data: img,
                        loading: false
                    })
                }, 3000)
            })
    }, [])

    // setTimeout(() => {
    //     setState({
    //         data: [1,2,3,4],
    //         loading: false
    //     })
    // }, 3000)

    return state;
}