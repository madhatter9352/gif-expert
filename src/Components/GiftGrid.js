import React from 'react'
import {GiftGridItem} from './GiftGridItem'

import { useFectchGifs } from '../hooks/useFetchGifs';

export const GiftGrid = ( { category } ) => {
    
    // const [images, setImages] = useState([]);
    const {data:images, loading} = useFectchGifs(category);
    // useEffect(() => {
    //     getGifs(category)
    //         .then( imgs => setImages(imgs))
    // }, [category])

    return (
        <>
        <h3>{category}</h3>
        { loading && <h1>Loading</h1> }
        <div className="card-grid">
            {images.map( gif => (
                    <GiftGridItem 
                        key={gif.id}
                        {... gif}
                    />
                ))}
        </div>
        </>
    )
}
