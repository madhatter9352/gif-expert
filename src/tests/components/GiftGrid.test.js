import { shallow } from "enzyme"
import { GiftGrid } from "../../Components/GiftGrid"
import { useFectchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en GiftGrid', () => {
    const category = "One peace";

    test('debe mostrar el componente correctamente', () => {
      
        const wraper = shallow( <GiftGrid category={category}/> )

        expect(wraper).toMatchSnapshot()
    })

    test('Debe mostrar items cuando se cargan imagenes', () => {
        
        const wraper = shallow( <GiftGrid category={category}/> )
    })
    
})
