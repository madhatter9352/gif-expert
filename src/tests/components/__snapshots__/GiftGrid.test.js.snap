// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftGrid debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    One peace
  </h3>
  <h1>
    Loading
  </h1>
  <div
    className="card-grid"
  />
</Fragment>
`;
